### PR TITLE
Fix --with-checked-build build on GCC/Linux

### PR DESCRIPTION
**Pull request by xmcclure**
My previous commits on the image audit feature work on clang (apparently defaults C99) but not gcc (apparently defaults C89). Oops. I think I also fix a bug in that last line.

### CODE REVIEW

File: mono/metadata/image.c [hunk 1]
```diff
@@ -2605,4 +2605,4 @@ mono_find_mempool_owner (void *ptr)
 	return owner;
 }
 
-#endif
\ No newline at end of file
+#endif
```
Comment (xmcclure) [inline mono/metadata/image.c]: vi

File: mono/metadata/image.c [hunk 2]
```diff
@@ -2578,10 +2578,10 @@ mono_find_mempool_owner (void *ptr)
 	gboolean searching = TRUE;
 
 	// Iterate over both by-path image hashes
-	const int hash_count = 2;
-	const int hash_candidates[hash_count] = {IMAGES_HASH_PATH, IMAGES_HASH_PATH_REFONLY};
+	#define HASH_CANDIDATE_COUNT 2
```
Comment (kumpera) [inline mono/metadata/image.c]: preprocessor # must be anchored on the first column.

File: mono/metadata/image.c [hunk 3]
```diff
@@ -2578,10 +2578,10 @@ mono_find_mempool_owner (void *ptr)
...
+	const int hash_candidates[HASH_CANDIDATE_COUNT] = {IMAGES_HASH_PATH, IMAGES_HASH_PATH_REFONLY};
 	int hash_idx;
-	for (hash_idx = 0; searching && hash_idx < IMAGES_HASH_COUNT; hash_idx++)
+	for (hash_idx = 0; searching && hash_idx < HASH_CANDIDATE_COUNT; hash_idx++)
```
Comment (alexrp) [inline mono/metadata/image.c]: Not familiar with the code, but this looks to be changing semantics while the commit itself just makes it sound like a build fix.

File: mono/metadata/image.c (hunk 1 above)
Comment (alexrp) [inline mono/metadata/image.c]: It's fine to add a newline at the end of files that don't have it so that we can avoid formatting noise from editors that add it automatically (which is most of them). Please split that into a separate commit, though.

**Comment by xmcclure** [comment]
Re-pushed, addressing above comments.

**Comment by xmcclure** [comment]
At recommendation of (comment lost??) switched to using `G_N_ELEMENTS` and I think this code is now ~~PERFECT~~

**Comment by alexrp** [comment]
LGTM
